### PR TITLE
Add tests for rust reqwest middleware client

### PR DESCRIPTION
**Pull request by wing328**
Add tests for rust reqwest middleware client

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @frol (2017/07) @farcaller (2017/08) @richardwhiuk (2019/07) @paladinzh (2020/05) @jacob-pro (2022/10)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1172,6 +1172,7 @@
                 <module>samples/client/petstore/rust</module>
                 <module>samples/client/petstore/rust/reqwest/petstore</module>
```
Comment (nshaaban-cPacket) [inline pom.xml:1173]: nit: Hyper directory is missing as well
```suggestion
                <module>samples/client/petstore/rust/hyper/petstore</module>
                <module>samples/client/petstore/rust/reqwest/petstore</module>
```

**Comment by jacob-pro** [comment]
I'm kind of confused why these need to be added individually like this?
All the crates exist within the same workspaces:
https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/rust/Cargo.toml
So if you run `cargo test` within the root of the workspace it will run all unit and integration tests in all the sub crates

File: .github/workflows/samples-rust.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          - samples/server/petstore/rust-server/output/openapi-v3
+          - samples/server/petstore/rust-server/output/ops-v3
+          - samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing
+          - samples/server/petstore/rust-server/output/ping-bearer-auth
+          - samples/server/petstore/rust-server/output/rust-server-test
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
```
Comment (jacob-pro) [inline .github/workflows/samples-rust.yaml:37]: We should consider setting an MSRV here, or ideally having a matrix of both MSRV and latest. Similar to:
https://github.com/actix/actix-web/blob/45b77c68195eb933231290a09e9f6a0cca56aad8/.github/workflows/ci.yml#L21-L23

We should only run clippy on the MSRV so we avoid the issues here:
https://github.com/OpenAPITools/openapi-generator/pull/13907

File: .github/workflows/samples-rust.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          - samples/client/petstore/rust/reqwest/petstore-async
+          - samples/client/petstore/rust/reqwest/petstore-awsv4signature
+          - samples/client/petstore/rust/reqwest/petstore
+          - samples/server/petstore/rust-server/output/multipart-v3
+          - samples/server/petstore/rust-server/output/no-example-v3
+          - samples/server/petstore/rust-server/output/openapi-v3
+          - samples/server/petstore/rust-server/output/ops-v3
+          - samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing
+          - samples/server/petstore/rust-server/output/ping-bearer-auth
+          - samples/server/petstore/rust-server/output/rust-server-test
```
Comment (jacob-pro) [inline .github/workflows/samples-rust.yaml:32]: Can this not just be replaced with?
```
samples/client/petstore/rust
samples/server/petstore/rust-server
```

Comment (wing328) [reply .github/workflows/samples-rust.yaml:32]: These are values of the variables `${{ matrix.sample }}`, which is used in the "working-directory" below in `Build`.

File: pom.xml (hunk 1 above)
Comment (wing328) [reply pom.xml:1173]: don't recall why it's omitted. agreed it should be included if the test passes.

File: .github/workflows/samples-rust.yaml (hunk 3 above)
Comment (jacob-pro) [reply .github/workflows/samples-rust.yaml:32]: Yeah I know, but I'm saying you don't need to manually go into each directory. You should just be able to run the build command in the workspace root (see my other comment)

File: .github/workflows/samples-rust.yaml (hunk 2 above)
Comment (wing328) [reply .github/workflows/samples-rust.yaml:37]: thanks for sharing that. agreed with the matrix of both MSRV and the latest.

let's see if anyone has time to do it in a separate PR after merging this one.

File: .github/workflows/samples-rust.yaml (hunk 3 above)
Comment (wing328) [reply .github/workflows/samples-rust.yaml:32]: Ah I see what you mean now. You're right. We can simply use the 2 lines you mentioned above.

**Pull request merged**